File: frontend/react/src/api/CategoryApi.ts
import axiosInstance from './axiosInstance';
import type { CategoryCreate, CategoryResponse } from '../types/category';

export const getCategories = async (storeId: number): Promise<CategoryResponse[]> => {
  const response = await axiosInstance.get<CategoryResponse[]>(`/categories/store/${storeId}`);
  return response.data;
};

export const createCategory = async (data: CategoryCreate): Promise<CategoryResponse> => {
  const response = await axiosInstance.post<CategoryResponse>('/categories', data);
  return response.data;
};

File: frontend/pyside6/api/category_api.py
import requests

BASE_URL = "http://127.0.0.1:8000"

def get_categories_by_store(store_id: int):
    """/categories/store/{store_id}"""
    
    response = requests.get(
        f"{BASE_URL}/categories/store/{store_id}"
    )
    response.raise_for_status()
    return response.json()

def create_category(store_id: int, name: str):
    response = requests.post(
        f"{BASE_URL}/categories",
        json={
            "store_id": store_id,
            "name": name
        }
    )
    response.raise_for_status()
    return response.json()